app/javascript/controllers/codesnippet_controller.js

```diff
@@ -0,0 +1,40 @@
+import { Controller } from "@hotwired/stimulus"
+
+export default class extends Controller {
+    static targets = ["clipboardText", "clipboardCheck"];
+
+    async copyToClipboard(event) {  
+        let text = this.copyActivePaneContent();
+        await navigator.clipboard.writeText(text);
+        this.displayCopyClick();
+    }
+
+    copyActivePaneContent() {
+      const codeText = document.querySelector("div#codeContent div.tab-pane.active pre code");
```

Comment:
can this be implemented at a stimulus target?  If not, can we use `this.element` rather than `document`?

Reply:
I added a target for div#codeContent (which is the nav tab section) and then used the query selector directly.